File: 2021년/9월/0907 피보나치 수 7.py
# 백준_15624_피보나치 수 7_DP_실버 3

n = int(input())
fibo = [0, 1] + [0]*(n-1)
for i in range(2, n+1):
    fibo[i] = (fibo[i-2] + fibo[i-1]) % 1000000007

print(fibo[n])

File: 2021년/9월/0907 피보나치 수 7.cpp
#include <iostream>
using namespace std;

int n;
long long fibo[1000001];

int main()
{
	scanf("%d", &n);
	fibo[1] = 1;
	for(int i=2;i<n+1;i++)
	{
		fibo[i] = (fibo[i-1] + fibo[i-2])%1000000007;
	}
	printf("%lld", fibo[n]);
}
